### DISCUSSION TITLE
How can I access the `index` within the carousel?

### DISCUSSION THREAD

**Discussion by austinmrobinson** [dscyrescotti/ShuffleIt#70] [Q&A] [answered]
I'd like to display the typical dot indicators in the carousel, but I think I'd need the index to be able to indicate the current item. I've looked at the docs and examples and can't seem to figure it out unfortunately

**Comment by austinmrobinson** [comment, marked as answer]
I figured it out by creating an index variable
`@State var carouselIndex: Int` and setting it within `onCarousel`
``` Swift
.onCarousel { context in
      carouselIndex = context.index
}
```

**Reply by dscyrescotti** [reply]
@AustinMRobinson Yes, you can use `onCarousel(_:)` modifier to listen to the index changes of carousel.
